src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -45,28 +46,27 @@ pub(super) struct LocationState {
     initialized: bool,
     /// This pointer's current permission / future initial permission.
     permission: Permission,
-    /// Strongest foreign access whose effects have already been applied to
-    /// this node and all its children since the last child access.
-    /// This is `None` if the most recent access is a child access,
-    /// `Some(Write)` if at least one foreign write access has been applied
-    /// since the previous child access, and `Some(Read)` if at least one
-    /// foreign read and no foreign write have occurred since the last child access.
-    latest_foreign_access: Option<AccessKind>,
+    /// See `foreign_access_skipping.rs`
+    /// Stores the strongest idempotent foreign access for this location.
+    /// Invariant is that at all children, the strongest idempotent foreign access is stronger (greater).
```

Comment:
This is an approximation, right? It's okay for this to always be `LocalAccess` even if it would be idempotent under foreign reads? That should be documented.

Reply:
It's always OK for this field to be `LocalAccess`. Also for the one in `Node`. That would be as if this whole optimization did not exist.

Reply:
Okay, then please explain that. Currently the definition of SIFA makes it sound like an exact bound.